src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -84,14 +84,132 @@ private static IEnumerable<TextEdit> FormatRazor(FormattingContext context, Razo
+                var code = cSharpCode.Children.First(c => c.Kind == SyntaxKind.CSharpCodeBlock) as CSharpCodeBlockSyntax;
+
+                if (openBrace is null || closeBrace is null)
+                {
+                    // Don't trust ourselves in an incomplete scenario.
+                    return false;
+                }
+
+                var openBraceNode = openBrace;
+                var codeNode = code!;
+                var closeBraceNode = closeBrace;
+
+                return FormatBlock(context, source, openBraceNode, codeNode, closeBraceNode, edits, additionalIndentation: false);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryFormatCSharpExplicitTransition(FormattingContext context, IList<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+        {
+            // We're looking for a code block like this:
+            //
+            // @{
+            //     var x = 1;
+            // }
+            if (node is CSharpCodeBlockSyntax expliciteCode &&
+                expliciteCode.Children.First() is CSharpStatementSyntax statement &&
+                statement.Body is CSharpStatementBodySyntax csharpStatementBody)
+            {
+                var openBraceNode = csharpStatementBody.OpenBrace;
+                var codeNode = csharpStatementBody.CSharpCode;
+                var closeBraceNode = csharpStatementBody.CloseBrace;
+
+                // IndentationContext doesn't track the need for any additional indentation in this case (Both Razor and HTML desirred indentation would be "0" above).
+                // Probably because it thinks you're doing something like "@{ DateTime.Now }"
+                // Regardless we need to help things out and inform FormatBlock to give us an extra level.
+                var additionalIndentation = true;
+
+                return FormatBlock(context, source, openBraceNode, codeNode, closeBraceNode, edits, additionalIndentation);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryFormatComplexCSharpBlock(FormattingContext context, IList<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+        {
+            // complex situations like
+            // @{
+            //  void Method(){
+            //      @(DateTime.Now)
+            //  }
+            // }
+            if (node is CSharpRazorBlockSyntax csharpRazorBlock &&
+                csharpRazorBlock.Parent is CSharpCodeBlockSyntax innerCodeBlock &&
+                csharpRazorBlock.Parent.Parent is CSharpCodeBlockSyntax outerCodeBlock)
+            {
+                var codeNode = csharpRazorBlock;
+                var openBraceNode = outerCodeBlock.Children.PreviousSiblingOrSelf(innerCodeBlock);
+                var closeBraceNode = outerCodeBlock.Children.NextSiblingOrSelf(innerCodeBlock);
```

Comment:
lol I'm sure the C# compiler team would shake their head at us seeing this code; little do they know it's a magical hack that actaully makes too much sense in Razor land. At least until the compiler has more structured concepts anyways 😄